runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -129,7 +129,6 @@ public synchronized void acquireBuffer() {
         if (serializedCache == null) {
             serializedCache = Unpooled.buffer();
             doSerializeInternal(serializedCache);
-            lastKnownSize = serializedCache.array().length;
```

Comment:
can't you just set lastKnownSize = buf.readableBytes() on acquireBuffer  instead of setting it twice in doSerializeInternal ?

Reply:
That's what I initially did, but later I realized that that way the lastKnownSize will include the metadata map, but when deserializing, it does include metadata.